src/raiden_libs/matrix.py

```diff
@@ -391,8 +387,20 @@ def connect_client_forever(self, server_url: str) -> None:
                 client = self._start_client(server_url)
                 assert client.sync_worker is not None
                 client.sync_worker.get()
-            except (TransportError, ConnectionError):
-                log.debug("Could not connect to server", server_url=server_url)
+            except (TransportError, ConnectionError) as ex:
+                log.error("Could not connect to server", server_url=server_url, exception=ex)
+            except MatrixRequestError as ex:
```

Comment:
This means that we assume `MatrixRequestError`s are transient. 
But I guess it's fine. If the server is out of action more long-term it's clients need to roam to a different server anyway.

Reply:
As this is the code base for the PFS to connect to all servers, this particular connection attempt should try as long as it lives even though the error might be persistent. Because the PFS is connected to all other homeservers with its own client there is no need to roam.

Reply:
Yeah, I meant the Raiden nodes on a more long term "broken" homeserver will eventually roam to a different one.

Reply:
I think thats the reason why a 404 is not handled in the MatrixClient